src/client/src/producer.rs

```diff
@@ -96,82 +200,103 @@ impl TopicProducer {
         skip(self, buffer),
         fields(topic = %self.topic),
     )]
+    #[deprecated(since = "0.6.2", note = "Please use 'send_keyless' instead")]
     pub async fn send_record<B: AsRef<[u8]>>(
         &self,
         buffer: B,
-        partition: i32,
+        _partition: i32,
     ) -> Result<(), FluvioError> {
-        let record = buffer.as_ref();
-        let replica = ReplicaKey::new(&self.topic, partition);
-        debug!("sending records: {} bytes to: {}", record.len(), &replica);
+        let buffer: Vec<u8> = Vec::from(buffer.as_ref());
+        self.send_keyless(buffer).await?;
+        Ok(())
+    }
+}
 
-        let spu_client = self.pool.create_serial_socket(&replica).await?;
+/// A trait for defining a partitioning strategy for key/value records.
+///
+/// A Partitioner is given a slice of potential keys, and the number of
+/// partitions in the current Topic. It must map each key from the input
+/// slice into a partition stored at the same index in the output Vec.
+///
+/// It is up to the implementor to decide how the keys get mapped to
+/// partitions. This includes deciding what partition to assign to records
+/// with no keys (represented by `None` values in the keys slice).
+///
+/// See [`SiphashRoundRobinPartitioner`] for a reference implementation.
+trait Partitioner {
+    fn partition(&mut self, key: Option<&[u8]>) -> i32;
```

Comment:
See previous comment, this should take on value as well

Reply:
- Which previous comment are you referring to?
- Why should the partitioner accept the value for the record? What is an example use-case that you are imagining?

Reply:
It was in previous comment:  https://kafka.apache.org/24/javadoc/?org/apache/kafka/clients/producer/Partitioner.html.  Kafka produce takes on value which make sense.   There is no need to limit to key only.

Reply:
Oh interesting, it looks like they have partitioning strategies that just ignore the key and look at the value contents to find a defined partition https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/UniformStickyPartitioner.java#L25-L36